File: mirai/edge/template/mirai_tools/kotlin/src/main/kotlin/io/mirai/edge/MiraiSetup.kt
package io.mirai.edge

import io.mirai.sdk.AgentOptions
import io.mirai.sdk.MiraiAgent
import io.mirai.sdk.RegisterOptions
import io.mirai.sdk.ToolHandler
import io.mirai.sdk.ToolParameter

fun initMirai() {
    val agent = MiraiAgent(
        AgentOptions(
            connectionCode = null,
            edgeName = "My Kotlin App",
            envPath = null,
        ),
    )
    agent.register(
        RegisterOptions(
            name = "hello",
            description = "Say hello to someone",
            parameters = listOf(
                ToolParameter(
                    name = "name",
                    typeName = "string",
                    description = "Person to greet",
                ),
            ),
            handler = ToolHandler { args -> "Hello, ${args.string("name")}!" },
        ),
    )
    agent.runInBackground()
}

File: mirai/edge/template/mirai_tools/rust/src/mirai_setup.rs
use mirai_sdk::{AgentOptions, MiraiAgent, RegisterOptions, ToolParameter};
use std::sync::Arc;

/// Call this from `main` or embed in your own async runtime.
pub fn init_mirai() {
    let agent = MiraiAgent::new(AgentOptions {
        connection_code: None,
        edge_name: Some("My Rust App".into()),
        env_path: None,
    });

    agent.register(RegisterOptions {
        name: "hello".into(),
        description: "Say hello to someone".into(),
        parameters: vec![ToolParameter {
            name: "name".into(),
            type_name: "string".into(),
            description: "Person to greet".into(),
            required: None,
        }],
        timeout: None,
        require_confirmation: false,
        always_include: false,
        handler: Arc::new(|args| {
            let name = args.string("name");
            if name.is_empty() {
                "Hello, World!".into()
            } else {
                format!("Hello, {name}!")
            }
        }),
    });

    agent.run_in_background();
}
